src/getPropertyByPath.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+'use strict';
+
+// Resolves property paths defined with period-delimited strings or arrays of strings.
+// Property names that include periods are only understand in array paths.
```

Comment:
Had a moment to browse now - this differs from the `_.get` implementation which special cases a string to first test for the key before splitting on a period.

For backward comaptibility (even before _.get was introduced) we could do the same e.g.
```js
// getPropertyByPath.js

// Resolves property paths defined with period-delimited strings or arrays of strings.
function getPropertyByPath(source: Object, path: string | Array<string>): any {
  // first attempt the path as though it were a single key regardless whether it has periods
  if (typeof path === 'string' && source.hasOwnProperty(path))
    return source[path];

  const parsedPath = typeof path === 'string' ? path.split('.') : path;
  let config = source;
  for (const key of parsedPath) {
    if (!config || !config.hasOwnProperty(key))
      return;

    config = config[key];
  }

  return config;
}
```

and

```js
// getPropertyByPath.test.js

const source = {
  'ant.man': 'superhero',
}

test('attempts the string key before splitting', () => {
  expect(getPropertyPath(source, 'ant.man')).toBe(source['ant.man']);
});
```

Reply:
Good point, @olsonpm. I added a commit to address this.